Automattic/wp-calypso
# Themes: Move upload dropzone to a separate block

Factor out the drop-zone and file handling from the theme uploads page so that it can be shared by the forthcoming plugin upload page in #16110.

This is strictly a refactor: no visual differences, all theme upload behaviour should match master.

<img width="777" alt="screen shot 2017-07-18 at 20 26 59" src="https://user-images.githubusercontent.com/7767559/28335791-7d6e0c02-6bf7-11e7-93d6-e8d92cd0c95c.png">

**To Test**
* Go to http://calypso.localhost:3000/themes/upload/{jetpack site}
* The file dropzone should behave exactly as in production
* Try: invalid files, multiple files, valid theme zip
* Also, ATomic transfers should continue to function, following instructions here: p58i-4Uy-p2
(I have tested this!)
